src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -274,16 +285,84 @@ def _deploy_arm_template_core(cli_ctx, resource_group_name,
+                    data.properties.template = None
+                    data_as_dict = data.serialize()
+                    data_as_dict["properties"]["template"] = JsonCTemplate(template)
+                    return data_as_dict
+            return super(MySerializer, self).body(data, data_type, **kwargs)
+    deployments_operation_group = smc.deployments  # This solves the multi-api for you
+
+    # pylint: disable=protected-access
+    deployments_operation_group._serialize = MySerializer(
+        deployments_operation_group._serialize.dependencies
+    )
+
+    # Now, you have a serializer that keeps a weird class inside it, you need to explain to the HTTP pipeline how to translate that into bytes
+    from msrest.pipeline import SansIOHTTPPolicy
+
+    class JsonCTemplatePolicy(SansIOHTTPPolicy):
+        def on_request(self, request, **kwargs):
+            http_request = request.http_request
+            print(http_request.data)
+            try:
+                template = http_request.data["properties"]["template"]
+                if not isinstance(template, JsonCTemplate):
+                    raise ValueError()
+            except (KeyError, ValueError):
+                # Not a template
+                return
+            # I know it's my weird template
+            del http_request.data["properties"]["template"]
+            # templateLink nad template cannot exist at the same time in deployment_dry_run mode
+            if "templateLink" in http_request.data["properties"].keys():
+                del http_request.data["properties"]["templateLink"]
+            partial_request = json.dumps(http_request.data)
+
+            http_request.data = partial_request[:-2] + ", template:" + template.template_as_bytes + r"}}"
+
+    # Plug this as default HTTP pipeline
+    from msrest.pipeline import Pipeline
+    from msrest.pipeline.requests import (
+        RequestsCredentialsPolicy,
+        RequestsPatchSession,
+        PipelineRequestsHTTPSender
+    )
+    from msrest.universal_http.requests import RequestsHTTPSender
+
+    smc.config.pipeline = Pipeline(
+        policies=[
+            JsonCTemplatePolicy(),
+            smc.config.user_agent_policy,
+            RequestsPatchSession(),
+            smc.config.http_logger_policy,
+            RequestsCredentialsPolicy(smc.config.credentials)
+        ],
+        sender=PipelineRequestsHTTPSender(RequestsHTTPSender(smc.config))
+    )
+
     if validate_only:
-        return sdk_no_wait(no_wait, smc.deployments.validate, resource_group_name, deployment_name, properties)
-    return sdk_no_wait(no_wait, smc.deployments.create_or_update, resource_group_name, deployment_name, properties)
+        return sdk_no_wait(no_wait, deployments_operation_group.validate, resource_group_name, deployment_name, properties)
+    return sdk_no_wait(no_wait, deployments_operation_group.create_or_update, resource_group_name, deployment_name, properties, template_content)
```

Comment:
`template_content` is not supposed to be used as the parameter value. I suggest you cross check the SDK method, which should reveal the `template_content` will be picked up as the `location`, and then service end will reject with 400

Reply:
Right. For `create` command, it will be picked up as `location` but for `validate` it is right in `properties` field. I will investigate it more deeply.